src/Normalizer/ContentEntityNormalizer.php

```diff
@@ -297,7 +297,7 @@ protected function getEntityUri(EntityInterface $entity) {
     // out to ->url().
     if ($entity->isNew() || !$entity->hasLinkTemplate('canonical')) {
       if ($entity->getEntityTypeId() == 'file') {
-        return $entity->url();
+        return $entity->toUrl();
```

Comment:
This WSODs on my files REST export view for testing the `FileEntityNormalizer`: 

`The website encountered an unexpected error. Please try again later.</br></br><em class="placeholder">Drupal\Core\Entity\Exception\UndefinedLinkTemplateException</em>: No link template &#039;canonical&#039; found for the &#039;file&#039; entity type in <em class="placeholder">Drupal\Core\Entity\EntityBase-&gt;toUrl()</em> (line <em class="placeholder">227</em> of <em class="placeholder">core/lib/Drupal/Core/Entity/EntityBase.php</em>). <pre class="backtrace">Drupal\jsonld\Normalizer\ContentEntityNormalizer-&gt;getEntityUri(Object) (Line: 154)`


Think it needs to be `$entity->createFileUrl(FALSE)` instead, https://www.drupal.org/node/3019830.

Reply:
my bad - i thought this was a node entity since it was in the ContentEntityNormalizer but now i see its wrapped if entity type is file. i'll push up a fix